cellprofiler/modules/fillobjects.py

```diff
@@ -0,0 +1,90 @@
+# coding=utf-8
+
+"""
+FillObjects
+===========
+
+**FillObjects** fills holes within all objects in an image.
+
+This module works best on integer-labeled images (i.e., the output of **ConvertObjectsToImage**
+when the color format is *uint16*).
+
+The output of this module is a labeled image of the same data type as the input.
+**FillObjects** can be run *after* any labeling or segmentation module (e.g.,
+**ConvertImageToObjects** or **Watershed**). Labels are preserved and, where possible, holes
+entirely within the boundary of labeled objects are filled with the surrounding object number.
+
+|
+
+============ ============ ===============
+Supports 2D? Supports 3D? Respects masks?
+============ ============ ===============
+YES          YES          NO
+============ ============ ===============
+
+"""
+
+import numpy
+import skimage.morphology
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+
+
+class FillObjects(cellprofiler.module.ImageProcessing):
```

Comment:
Typically, modules with "Objects" in their name operate on CellProfiler `Objects`, rather than a CellProfiler `Image`. The `Objects` have a property `segmented` which returns the label matrix (essentially, the equivalent of **ConvertObjectsToImage** when the color format is *uint16*). 

With few modifications, this module could be reworked to operate on `Objects` by inheriting from `cellprofiler.module.ObjectProcessing`. This would save users the extra steps of converting their objects to and image, then back again. Measurements relating the original objects to the filled objects would also be recorded. Additionally, this module won't have to worry about whether or not it was passed the correct type of image data. 

What do you think?

Reply:
I am all for it! The other implementations for object operations (`SplitOrMergeObjects, FilterObjects`) looked quite complex when I glanced over them, and I wanted to make sure that the CellProfiler team felt this module would actually be beneficial to the software before spending a ton of time on it. I'll work to refactor the module to operate on `Objects` instead (as the name already implied :stuck_out_tongue: ).